app/src/main/java/com/example/internetshop/presentation/activity/fragments/CategoryProductListFragment.kt

```diff
@@ -0,0 +1,82 @@
+package com.example.internetshop.presentation.activity.fragments
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.os.bundleOf
+import androidx.fragment.app.viewModels
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.internetshop.databinding.FragmentCategoryProductListBinding
+import com.example.internetshop.model.data.di.component.AppComponent
+import com.example.internetshop.presentation.activity.ContainerHolder
+import com.example.internetshop.presentation.adapters.ProductsCategoryAdapter
+import com.example.internetshop.presentation.adapters.VerticalSpaceItemDecoration
+import com.example.internetshop.presentation.viewModel.CategoryProductListViewModel
+
+class CategoryProductListFragment : BaseFragment() {
+
+    private var binding: FragmentCategoryProductListBinding? = null
+
+    val viewModel: CategoryProductListViewModel by viewModels { factory }
+
+    companion object {
+        const val EXTRA_CATEGORY_NAME = "category"
+    }
+
+    override fun inject(component: AppComponent) {
+        component.inject(this)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+        super.onCreateView(inflater, container, savedInstanceState)
+        binding = FragmentCategoryProductListBinding.inflate(inflater, container, false)
+        return binding?.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        val adapter = ProductsCategoryAdapter { viewModel.onClicked(it) }
+        val recyclerView = binding?.recyclerViewProducts
+        recyclerView?.let {
+            it.addItemDecoration(VerticalSpaceItemDecoration(32))
+            it.adapter = adapter
+            it.layoutManager = LinearLayoutManager(requireContext())
+            it.setHasFixedSize(true)
+        }
+
+        viewModel.productsList.observe(requireActivity(), {
+            adapter.addData(it)
+            adapter.notifyDataSetChanged()
+        })
+        val categoryName = this.requireArguments().getString(EXTRA_CATEGORY_NAME)!!
```

Comment:
Possible crash here. If extra is null then you will get `NullPointerException`. Would be better to handle possible null value instead of using non-safe `!!`.

Reply:
fixed